pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_drop_column.go

```diff
@@ -391,6 +390,56 @@ func walkDropColumnDependencies(b BuildCtx, col *scpb.Column, fn func(e scpb.Ele
 	})
 }
 
+// panicIfColReferencedInPredicate is a temporary fix that disallow dropping a
+// column that is referenced in predicate of a partial index or unique without index.
+// This restriction shall be lifted once #96924 is fixed.
+func panicIfColReferencedInPredicate(b BuildCtx, col *scpb.Column, tblElts ElementResultSet) {
+	contains := func(container []catid.ColumnID, target catid.ColumnID) bool {
+		for _, elem := range container {
+			if elem == target {
+				return true
+			}
+		}
+		return false
+	}
+
+	var violatingIndex catid.IndexID
+	var violatingUWI catid.ConstraintID
+	tblElts.ForEachElementStatus(func(_ scpb.Status, _ scpb.TargetStatus, e scpb.Element) {
+		if violatingIndex != 0 || violatingUWI != 0 {
+			return
+		}
+		switch elt := e.(type) {
+		case *scpb.SecondaryIndex:
+			if elt.EmbeddedExpr != nil && contains(elt.EmbeddedExpr.ReferencedColumnIDs, col.ColumnID) {
+				violatingIndex = elt.IndexID
```

Comment:
nit: looks like we may have multiple violating indexes, but we only panic only the last one?

Reply:
This is only for the error message, and I think it's okay that we only include one.

Reply:
sgtm